zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/changes/PartitionBootstrapApplier.java

```diff
@@ -80,14 +97,17 @@ public Either<Exception, UnaryOperator<MemberState>> initMemberState(
                   .formatted(partitionId)));
     }
 
-    // Let's assume Partition 1 always exists
     partitionConfig =
-        currentClusterConfiguration.members().values().stream()
-            .flatMap(m -> m.partitions().entrySet().stream().filter(p -> p.getKey() == 1))
-            .findFirst()
-            .get()
-            .getValue()
-            .config();
+        (config.orElse(
```

Comment:
:wrench: Let's extract the fallback behavior to a method to make it clearer what we're doing. Added as a follow up in the main issue.